docs/proposals/formatted-output.md

```diff
@@ -379,10 +379,133 @@ oras discover $REGISTRY/$REPO:v1 --format json
+          },
+          "artifactType": "application/vnd.cncf.notary.signature"
+        }
+      ]
+    },
+    {
+      "reference": "localhost:5000/kubernetes/kubectl@sha256:a811606b09341bab4bbc0a4deb2c0cb709ec9702635cbe2d36b77d58359ec046",
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:a811606b09341bab4bbc0a4deb2c0cb709ec9702635cbe2d36b77d58359ec046",
+      "size": 747,
+      "annotations": {
+        "org.opencontainers.image.created": "2024-01-18T18:12:41Z"
+      },
+      "artifactType": "application/vnd.in-toto+json",
+      "referrers": [
+        {
+          "reference": "localhost:5000/kubernetes/kubectl@sha256:04723fd7d00df77c6f226b907667396554bf9418dc48a7a04feb5ff24aa0b9ec",
+          "mediaType": "application/vnd.oci.image.manifest.v1+json",
+          "digest": "sha256:04723fd7d00df77c6f226b907667396554bf9418dc48a7a04feb5ff24aa0b9ec",
+          "size": 1080,
+          "annotations": {
+            "io.cncf.notary.x509chain.thumbprint#S256": "[430ecf0685f8018443f8418f5d7134b146f28862116114925713635d5703fb69,9b1894f223d934cbd6575af3c6e1f6096b9221a7da132185f5a5cdc92235b5dc,23ffe2b8bdb9a1711515d4cffda04bc7f793d513c76c243f1020507d8669b7db]",
+            "org.opencontainers.image.created": "2024-01-18T18:20:09Z"
+          },
+          "artifactType": "application/vnd.cncf.notary.signature"
+        }
+      ]
+    }
+  ]
+```
+
+#### Set the depth of listed referrers
+
+ORAS shows all referrers of a subject image by default. To avoid throttling or hitting a performance issue when a subject image has a complicated graph of referrers, ORAS introduces an experimental flag `--depth` to `oras discover` to allow users to set the maximum depth of referrers in the formatted output. 
+
+For example, assume there is a sample image with four-level referrers, show referrers within the thrid level only in a tree view:
+
+```bash
+oras discover localhost:5000/kubernetes/kubectl@sha256:bece4f4746a39cb39e38451c70fa5a1e5ea4fa20d4cca40136b51d9557918b01 --format tree --depth 3
+```
+
+```console
+localhost:5000/kubernetes/kubectl@sha256:bece4f4746a39cb39e38451c70fa5a1e5ea4fa20d4cca40136b51d9557918b01
+├── application/vnd.oci.artifact.lifecycle
+│   └── sha256:325129be79f416fe11a9ec44233cfa57f5d89434e6d37170f97e48f7904983e3
+│       └── application/vnd.cncf.notary.signature
+│           └── sha256:f520330e9f43c05859c532e67a25c9c765b144782ae7b872656192c27fd4e2dd
+│               └── vnd/test-annotations
+│                   └── sha256:d2cb66a53e4d77488df1f15701554ebb11ffa1cf6eb90f79afa33d3b172f11d2
+└── application/vnd.in-toto+json
+    └── sha256:a811606b09341bab4bbc0a4deb2c0cb709ec9702635cbe2d36b77d58359ec046
+        └── application/vnd.cncf.notary.signature
+            └── sha256:04723fd7d00df77c6f226b907667396554bf9418dc48a7a04feb5ff24aa0b9ec
+                └── vnd/test-annotations
+                    └── sha256:d2cb66a53e4d77488df1f15701554ebb11ffa1cf6eb90f79afa33d3b172f11d2
+```
+
```

Comment:
We are going to deprecate the table view. This should be included in the spec.

Reply:
Updated @wangxiaoxuan273